### PR TITLE
templater: use .map_or() to silence clippy without sacrificing readability

### PR DESCRIPTION
**Pull request by yuja**
# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [ ] I have added tests to cover my changes

### CODE REVIEW

**Review by PhilipMetzger — APPROVED** [review]

**Review by ilyagr — APPROVED** [review]
Thanks, this seems to work! Looks better too.

Of course, it's only a matter of time before clippy suggests replacing that with `unwrap_or` :)

File: cli/src/commit_templater.rs [hunk 1]
```diff
@@ -1027,14 +1027,8 @@ impl RefNamesIndex {
...
 
-    #[allow(unknown_lints)] // XXX FIXME (aseipp): nightly bogons; re-test this occasionally
-    #[allow(clippy::manual_unwrap_or_default)]
     pub fn get(&self, id: &CommitId) -> &[Rc<RefName>] {
-        if let Some(names) = self.index.get(id) {
-            names
-        } else {
-            &[]
-        }
+        self.index.get(id).map_or(&[], |names| names)
```
Comment (ilyagr) [inline cli/src/commit_templater.rs:1031]: Optional: add a comment why `unwrap_or` does not work here? `|names| names` does look slightly awkward.

Also, I wouldn't mind understanding the issue better myself.

It's fine if you don't feel like it.

**Comment by ilyagr** [comment]
Cc: https://github.com/rust-lang/rust-clippy/issues/13018

File: cli/src/commit_templater.rs (hunk 1 above)
Comment (yuja) [reply cli/src/commit_templater.rs]: Ok, I added type annotation `names: &Vec<_>` to better explain `|x| x`.

**Pull request merged**